.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,33 @@
+## Feature Title[Provide a succinct and clear title for the feature, e.g., "Introduce advanced search functionality"]
+
+## Type of Change
+- [X] New feature (non-breaking change which adds functionality)
+- [ ] Fixes Bug (Fixes an existing functionality which was not working as expected)
+- [ ] Documentation change (Changes to Readme.md)
+
+## Description of the Feature
+[A brief overview of the feature being added. Explain the functionality and its intended purpose.]
+
+## Implementation Details
+[Describe your approach to implementing the feature. Include any design decisions and methods used.]
+
+## Impact of the Feature
+[Discuss how this feature affects the existing codebase, including any potential behavioural or performance changes.]
+
+## Testing Conducted
+[Detail the tests carried out to ensure the feature works as intended and integrates well with the existing system.]
```

Comment:
I think this many questions are overkill. Just keep one field called description.
PLUS, we need a field to link issue, that is super mandatory.

Reply:
Ill keep feature ,implementation and testing

Reply:
It will help in faster PR reviews

Reply:
For now, let's go with simplicity. We can update if we find challenges later on.